t/03.scripts/tweak_pipeline_output.t

```diff
@@ -0,0 +1,113 @@
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].resource_class#", "-json"];
+
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].is_excluded?", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].is_excluded=20", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].is_excluded#", "-json"];
+
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].some_wrong_attr?", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].some_wrong_attr=20", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].some_wrong_attr#", "-json"];
+
+#Check analysis.wait_for (show, delete, set)
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].wait_for=perform_cmd", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].wait_for?", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].wait_for#", "-json"];
+
+#Check analysis.flow_into (show, delete, set)
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].flow_into=perform_cmd", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].flow_into?", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].flow_into#", "-json"];
+
+#Check analysis.param (show, delete, set)
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].param[base]=10", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].param[base]?", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].param[base]#", "-json"];
+
+#Check resource_class (show, set, error)
+push @tweak_requests,  ["-tweak" => "resource_class[urgent].LSF=-q yesteryear", "-json"];
+push @tweak_requests,  ["-tweak" => "resource_class[urgent].LSF?", "-json"];
+
+foreach my $request (@tweak_requests) {
+    my $stdout = capture_stdout {
+        tweak_pipeline($pipeline_url, $request);
+    };
+    is_valid_json $stdout;
+    my $stdoutJson = decode_json($stdout);
+    use Data::Dumper;
+    ok(scalar @{$stdoutJson->{Tweaks}} > 0, "Tweaks responce recieved for " . join (' ', @{$request}) . Dumper($stdoutJson->{Tweaks}));
+    foreach my $tweakJson (@{$stdoutJson->{Tweaks}}) {
+        ok($tweakJson->{Object}->{Type} eq "Pipeline" ||
+            $tweakJson->{Object}->{Type} eq "Analysis" ||
+            $tweakJson->{Object}->{Type} eq "Resource class",
+            'Object type is correct for ' . join (' ', @{$request}));
+
+        ok( $tweakJson->{Action} eq "SET" ||
+            $tweakJson->{Action} eq "SHOW" ||
+            $tweakJson->{Action} eq "DELETE",
+            'Action field is correct for ' . join (' ', @{$request}));
+
+        ok(exists ($tweakJson->{Object}->{Id}), 'Id field exists for' . join (' ', @{$request}));
+        ok(exists ($tweakJson->{Object}->{Name}), 'Name field exists for' . join (' ', @{$request}));
+        ok(exists ($tweakJson->{Return}->{OldValue}) || $tweakJson->{Warning} || $tweakJson->{Error}, 'Old value exists for ' . join (' ', @{$request}));
```

Comment:
Can some of these `||` be made exclusive ?

Reply:
Can you change the mark in `@tweak_requests` which ones are expected to fail and test them appropriately here ? Otherwise, all tweaks could be returning an error and the unit-test would still be happy